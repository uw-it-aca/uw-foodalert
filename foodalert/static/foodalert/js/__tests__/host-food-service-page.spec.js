"use strict";
import Vue from 'vue';
import BootstrapVue from 'bootstrap-vue'
Vue.use(BootstrapVue);

import {mount, shallowMount} from '@vue/test-utils';
import FoodServicePage from '../pages/host/food-service-page.vue';

const $router = {
    push: jest.fn(),
}

describe('Food Categories Page', function () {
        test('renders correctly', function () {
            var wrapper = mount(FoodServicePage, {
            });
            expect(wrapper.element).toMatchSnapshot();
        });
	
	test('testing options - no option selected', function () {
	    var wrapper = mount(FoodServicePage, {
	        mocks: { $router }
	    });
	    const button = wrapper.findAll('button').at(0)
	    button.trigger('click')
	});

	test('testing options - first option selected', function () {
	    var wrapper = mount(FoodServicePage, {
	        mocks: { $router }
	    });
            const checkboxes = wrapper.findAll('input[type="checkbox"]') 
	    checkboxes.at(0).setChecked()
	    const button = wrapper.findAll('button').at(0)
	    button.trigger('click')
	});

	test('testing options - second option selected', function () {
	    var wrapper = mount(FoodServicePage, {
	        mocks: { $router }
	    });
            const checkboxes = wrapper.findAll('input[type="checkbox"]') 
	    checkboxes.at(1).setChecked()
	    const button = wrapper.findAll('button').at(0)
	    button.trigger('click')
	});

	test('testing options - none option selected', function () {
	    var wrapper = mount(FoodServicePage, {
	        mocks: { $router }
	    });
            const checkboxes = wrapper.findAll('input[type="checkbox"]') 
	    checkboxes.at(2).setChecked()
	    const button = wrapper.findAll('button').at(0)
	    button.trigger('click')
	});

	test('testing options - none option selected when other options were selected', function () {
	    var wrapper = mount(FoodServicePage, {
	        mocks: { $router }
	    });
            const checkboxes = wrapper.findAll('input[type="checkbox"]') 
	    checkboxes.at(0).setChecked()
	    checkboxes.at(1).setChecked()
	    checkboxes.at(2).setChecked()
	    const button = wrapper.findAll('button').at(0)
	    button.trigger('click')
	});

	test('back button', function () {
	    var wrapper = mount(FoodServicePage, {
	        mocks: { $router }
	    });
	    const button = wrapper.findAll('button').at(1)
	    button.trigger('click')
	});
})
