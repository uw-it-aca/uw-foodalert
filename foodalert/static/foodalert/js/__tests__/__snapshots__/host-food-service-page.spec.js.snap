// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Food Categories Page renders correctly 1`] = `
<div
  class="page"
>
  <div
    id="standard-notification"
  >
    <div
      class="collapse"
      id="notif-container"
      style="display: none;"
    >
      <div
        class="text-center py-3 container-fluid"
      >
        <div
          class="container"
        >
          <div
            class="row justify-content-center"
          >
            <div
              class="col-md-8"
            >
              <div
                class="h5 text-white pt-2"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
   
  <div
    class="page-content"
  >
     
    <div
      class="container"
    >
      <div
        class="row justify-content-center"
      >
        <div
          class="page-content-padding col-md-8"
        >
          <h1
            class="pt-4 pb-2"
            id="standard-heading"
          >
            
        Food service information
    
          </h1>
           
          <div
            id="standard-body"
          >
            <p>
              
            Let’s determine whether your food can be shared with the public. Please select all that apply.
        
            </p>
             
            <fieldset
              class="form-group"
            >
              <!---->
              <div
                role="group"
                tabindex="-1"
              >
                <div
                  class="custom-control custom-checkbox"
                >
                  <input
                    autocomplete="off"
                    class="custom-control-input"
                    type="checkbox"
                    value="preparedByAuth"
                  />
                  <label
                    class="custom-control-label"
                  >
                    <span>
                      
                    My food was prepared by UW Housing & Food Services or Bay Laurel Catering.
                
                    </span>
                  </label>
                </div>
                 
                <div
                  class="mt-3 custom-control custom-checkbox"
                >
                  <input
                    autocomplete="off"
                    class="custom-control-input"
                    type="checkbox"
                    value="hasPermit"
                  />
                  <label
                    class="custom-control-label"
                  >
                    <span>
                      
                    I have a UW Temporary Food Service Permit.
                    
                      <a
                        aria-controls="perm-info"
                        aria-expanded="false"
                        class=""
                        herf="#"
                        href="#"
                        role="button"
                        target="_self"
                      >
                         Learn more
                      </a>
                    </span>
                  </label>
                </div>
                 
                <div
                  class="custom-collapse collapse-text-box collapse"
                  id="perm-info"
                  style="display: none;"
                >
                  <div
                    class="custom-text-box"
                  >
                    
                When providing food to the public (anyone beyond staff/faculty of your unit), UW offices are required to secure a Temporary Food Permit through 
                    <a
                      href="#"
                    >
                      UW Environmental Health & Safety
                    </a>
                     to help ensure food service providers meet safety reulation and the food itself is safe for consumption.
            
                  </div>
                </div>
                 
                <div
                  class="mt-3 custom-control custom-checkbox"
                >
                  <input
                    autocomplete="off"
                    class="custom-control-input"
                    type="checkbox"
                    value="none"
                  />
                  <label
                    class="custom-control-label"
                  >
                    <span>
                      
                    None of these apply.
                
                    </span>
                  </label>
                </div>
                <!---->
                <!---->
                <!---->
              </div>
            </fieldset>
          </div>
           
          <div
            style="margin-top: 32px;"
          >
            <div
              class="row justify-content-between"
            >
              <div
                class="col-md-4 col-lg-3 order-md-2"
              >
                <button
                  class="btn mb-3 btn-primary btn-lg btn-block disabled"
                  disabled="disabled"
                  type="submit"
                >
                  Continue
                </button>
              </div>
               
              <div
                class="col-md-4 col-lg-3 order-md-1"
              >
                <button
                  class="btn hh-back-button btn-outline-secondary btn-lg btn-block"
                  type="submit"
                >
                  Back
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
   
  <footer
    class="text-center"
    id="relative-footer"
  >
    <a
      href="mailto:help@uw.edu?subject=Hungry Husky support"
    >
      Contact support
    </a>
     
    <p>
      © 2019 University of Washington
    </p>
  </footer>
</div>
`;
